scripts/install

```diff
@@ -1,80 +1,151 @@
+# CREATE DIRECTORIES
+#=================================================
 
-# Make directories and set rights
 sudo mkdir -p /home/yunohost.transmission/{progress,completed,watched}
+
+#=================================================
+# SECURING FILES AND DIRECTORIES
+#=================================================
+
 sudo chown -R debian-transmission:www-data /home/yunohost.transmission/
-sudo chown -R debian-transmission:debian-transmission /home/yunohost.transmission/progress /home/yunohost.transmission/watched
-sudo find /home/yunohost.transmission/ -type f | while read LINE; do sudo chmod 640 "$LINE" ; done
-sudo find /home/yunohost.transmission/ -type d | while read LINE; do sudo chmod 750 "$LINE" ; done
-sudo find /home/yunohost.transmission/watched -type d | while read LINE; do sudo chmod 770 "$LINE" ; done
-ynh_app_setting_set "$app" watchdir "/home/yunohost.transmission/watched"
+sudo chown -R debian-transmission: /home/yunohost.transmission/{progress,watched}
+sudo chmod -R 764 /home/yunohost.transmission
+sudo chmod -R 777 /home/yunohost.transmission/watched
+
+#=================================================
+# CONFIGURE TRANSMISSION
+#=================================================
+
+# Transmission has to be stopped before modifying its config
+sudo systemctl stop transmission-daemon
+
+# Create a RPC password
+rpcpassword=$(ynh_string_random)
+ynh_app_setting_set $app rpcpassword "$rpcpassword"
+
+ynh_replace_string "__RPC_PASSWORD_TO_CHANGE__" "$rpcpassword" ../conf/settings.json
+ynh_replace_string "__PATH__" "$path_url" ../conf/settings.json
+ynh_replace_string "__PEER_PORT__" "$peer_port" ../conf/settings.json
+ynh_replace_string "__PORT__" "$port" ../conf/settings.json
+sudo cp ../conf/settings.json /etc/transmission-daemon/settings.json
+
+#=================================================
+# STORE THE CHECKSUM OF THE CONFIG FILE
+#=================================================
+
+ynh_store_file_checksum /etc/transmission-daemon/settings.json
+
+#=================================================
+# YUNOHOST MULTIMEDIA INTEGRATION
+#=================================================
 
-# YunoHost multimedia
 wget -qq https://github.com/YunoHost-Apps/yunohost.multimedia/archive/master.zip
 unzip -qq master.zip
 sudo ./yunohost.multimedia-master/script/ynh_media_build.sh
 # Set rights on transmission directory (parent need to be readable by other, and progress need to be writable by multimedia. Because files will move)
 sudo ./yunohost.multimedia-master/script/ynh_media_addfolder.sh --source="/home/yunohost.transmission" --dest="share/Torrents"
 # And share completed directory
 sudo ./yunohost.multimedia-master/script/ynh_media_addfolder.sh --source="/home/yunohost.transmission/completed" --dest="share/Torrents"
+# Share also watched directory, to allow to use it easily
+sudo ./yunohost.multimedia-master/script/ynh_media_addfolder.sh --source="/home/yunohost.transmission/watched" --dest="share/Torrent to download"
 
```

Comment:
Hum, I guess you would say the lines 115 to 117 ?
The following lines are specific to transmission.

Or, you would mean an abstract helper for simplify this command ?